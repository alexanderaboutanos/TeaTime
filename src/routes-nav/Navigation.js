/** @format */

import { useContext } from "react";
import { Navbar, Container, Nav } from "react-bootstrap";
import UserContext from "../auth/UserContext";

const Navigation = ({ logout }) => {
  const { currentUser } = useContext(UserContext);

  function loggedInNav() {
    return (
      <Nav className="me-auto">
        <Nav.Link href="/my-teas">My Teas</Nav.Link>
        <Nav.Link href="/wish-list">Wish List</Nav.Link>
        <Nav.Link href="/discover">Discover</Nav.Link>
        <Nav.Link href="/profile">Profile</Nav.Link>
        <Nav.Link onClick={logout} href="/logout">
          Logout
        </Nav.Link>
      </Nav>
    );
  }

  function loggedOutNav() {
    return (
      <Nav className="me-auto">
        <Nav.Link href="/login">Login</Nav.Link>
        <Nav.Link href="/signout">Signup</Nav.Link>
      </Nav>
    );
  }

  return (
    <>
      <Navbar bg="dark" variant="dark">
        <Container>
          <Navbar.Brand href="/">TeaTime</Navbar.Brand>
          {currentUser ? loggedInNav() : loggedOutNav()}
        </Container>
      </Navbar>
    </>
  );
};

export default Navigation;
