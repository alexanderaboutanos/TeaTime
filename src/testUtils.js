/** @format */

import React from "react";
import UserContext from "./auth/UserContext";

const demoUser = {
  username: "testuser1",
  first_name: "Test",
  last_name: "User",
};

const UserProvider = ({ children, currentUser = demoUser }) => (
  <UserContext.Provider value={{ currentUser }}>
    {children}
  </UserContext.Provider>
);

export { UserProvider };
