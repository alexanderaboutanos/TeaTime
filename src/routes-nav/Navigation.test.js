/** @format */

import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import Navigation from "./Navigation";
import { UserProvider } from "../testUtils";

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <UserProvider>
        <Navigation />
      </UserProvider>
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <Navigation />
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot when logged out", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider currentUser={null}>
        <Navigation />
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("includes proper links when logged out", function () {
  const { queryByText } = render(
    <MemoryRouter>
      <UserProvider currentUser={null}>
        <Navigation />
      </UserProvider>
    </MemoryRouter>
  );

  // we should see the homepage
  const link1 = queryByText("TeaTime");
  expect(link1).toBeInTheDocument();

  // we should see the login, the signup, and help.
  const link2 = queryByText("Login");
  const link3 = queryByText("Signup");
  const link4 = queryByText("Help");
  expect(link2).toBeInTheDocument();
  expect(link3).toBeInTheDocument();
  expect(link4).toBeInTheDocument();

  // we should NOT see the rest of the links.
  const link5 = queryByText("My Teas");
  const link6 = queryByText("Wishlist");
  const link7 = queryByText("Discover");
  const link8 = queryByText("Profile");
  const link9 = queryByText("Logout");
  expect(link5).not.toBeInTheDocument();
  expect(link6).not.toBeInTheDocument();
  expect(link7).not.toBeInTheDocument();
  expect(link8).not.toBeInTheDocument();
  expect(link9).not.toBeInTheDocument();
});

it("includes proper links when logged in", function () {
  const { queryByText } = render(
    <MemoryRouter>
      <UserProvider>
        <Navigation />
      </UserProvider>
    </MemoryRouter>
  );

  // we should see the homepage
  const link1 = queryByText("TeaTime");
  expect(link1).toBeInTheDocument();

  // we should NOT see the login, the signup
  const link2 = queryByText("Login");
  const link3 = queryByText("Signup");
  expect(link2).not.toBeInTheDocument();
  expect(link3).not.toBeInTheDocument();

  // we should see the rest of the links.
  const link4 = queryByText("Help");
  const link5 = queryByText("My Teas");
  const link6 = queryByText("Wishlist");
  const link7 = queryByText("Discover");
  const link8 = queryByText("Profile");
  const link9 = queryByText("Logout");
  expect(link4).toBeInTheDocument();
  expect(link5).toBeInTheDocument();
  expect(link6).toBeInTheDocument();
  expect(link7).toBeInTheDocument();
  expect(link8).toBeInTheDocument();
  expect(link9).toBeInTheDocument();
});
