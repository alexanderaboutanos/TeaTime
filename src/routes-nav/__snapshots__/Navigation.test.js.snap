// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand navbar-dark bg-dark"
    style="z-index: 2;"
  >
    <div
      class="container"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        TeaTime
      </a>
      <div
        class="me-auto navbar-nav"
      >
        <a
          class="nav-link"
          data-rr-ui-event-key="/my-teas"
          href="/my-teas"
        >
          My Teas
        </a>
        <a
          class="nav-link"
          data-rr-ui-event-key="/wish-list"
          href="/wish-list"
        >
          Wish List
        </a>
        <a
          class="nav-link"
          data-rr-ui-event-key="/discover"
          href="/discover"
        >
          Discover
        </a>
        <a
          class="nav-link"
          data-rr-ui-event-key="/profile"
          href="/profile"
        >
          Profile
        </a>
        <a
          class="nav-link"
          data-rr-ui-event-key="/help"
          href="/help"
        >
          Help
        </a>
        <a
          class="nav-link"
          data-rr-ui-event-key="/"
          href="/"
        >
          Logout
        </a>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;

exports[`matches snapshot when logged out 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand navbar-dark bg-dark"
    style="z-index: 2;"
  >
    <div
      class="container"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        TeaTime
      </a>
      <div
        class="me-auto navbar-nav"
      >
        <a
          class="nav-link"
          data-rr-ui-event-key="/login"
          href="/login"
        >
          Login
        </a>
        <a
          class="nav-link"
          data-rr-ui-event-key="/signup"
          href="/signup"
        >
          Signup
        </a>
        <a
          class="nav-link"
          data-rr-ui-event-key="/help"
          href="/help"
        >
          Help
        </a>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
